contracts/eosiolib/action.hpp

```diff
@@ -50,7 +50,11 @@ namespace eosio {
       size_t size = action_data_size();
       char* buffer = (char*)( max_stack_buffer_size < size ? malloc(size) : alloca(size) );
```

Comment:
Add:
const bool heap_allocation = max_stack_buffer_size < size;
and use heap_allocation in both places.

Reply:
Thanks for the suggestion. It makes it clearer to read :)
I've updated the PR